arcgispro/3D Data Co-Registration.pyt

```diff
@@ -353,27 +353,86 @@ class Register_MultiType(object):
         config = dataclasses.asdict(codem_run_config)
-
+        
+        #Add output to details pane with parameters and their values
+        arcpy.AddMessage("=============PARAMETERS=============")
+        for print_parameter in config:
+            arcpy.AddMessage(f"{print_parameter}={str(config[print_parameter])}")
         arcpy.SetProgressorLabel("Step 1/4: Prepping AOI and Foundation Data")
         arcpy.SetProgressorPosition()
+        arcpy.AddMessage("=============PREPROCESSING DATA=============")
         fnd_obj, aoi_obj = codem.preprocess(config)
 
         fnd_obj.prep()
         aoi_obj.prep()
 
+        if fnd_obj.units:
+            arcpy.AddMessage(f"Linear unit for Foundation-{fnd_obj.type.upper()} detected as {fnd_obj.units}")
+        else:
+            arcpy.AddMessage(f"Linear unit for Foundation-{fnd_obj.type.upper()} not detected -> "
+                    "meters assumed")
+        arcpy.AddMessage(f"Calculated native resolution of {fnd_obj.type.upper()} as: "
+            f"{fnd_obj.native_resolution:.1f} meters")
+
+        if aoi_obj.units:
+            arcpy.AddMessage(f"Linear unit for AOI-{aoi_obj.type.upper()} detected as {aoi_obj.units}")
+        else:
+            arcpy.AddMessage(f"Linear unit for Foundation-{aoi_obj.type.upper()} not detected -> "
+                    "meters assumed")
+        arcpy.AddMessage(f"Calculated native resolution of {aoi_obj.type.upper()} as: "
+            f"{aoi_obj.native_resolution:.1f} meters")
+
+        arcpy.AddMessage(f"Preparing Foundation {fnd_obj.type.upper()} for registration.")
+        arcpy.AddMessage(f"Extracting DSM from FND {fnd_obj.type.upper()} with resolution of: "
+            f"{fnd_obj.resolution} meters")
+
+        arcpy.AddMessage(f"Preparing AOI {aoi_obj.type.upper()} for registration.")
+        arcpy.AddMessage(f"Extracting DSM from AOI {aoi_obj.type.upper()} with resolution of: "
+            f"{aoi_obj.resolution} meters")
+        #No resampling required message?
+        arcpy.AddMessage(f"Registration resolution has been set to: "
+            f"{fnd_obj.resolution} meters")
+
         arcpy.SetProgressorLabel("Step 2/4: Solving Coarse Registration")
         arcpy.SetProgressorPosition()
+        arcpy.AddMessage("=============BEGINNING COARSE REGISTRATION=============")
+
         dsm_reg = codem.coarse_registration(fnd_obj, aoi_obj, config)
+        arcpy.AddMessage("Solving DSM feature registration.")
+        arcpy.AddMessage(f"{len(dsm_reg.fnd_kp)} keypoints detected in foundation.")
+        arcpy.AddMessage(f"{len(dsm_reg.aoi_kp)} keypoints detected in AOI.")
+        arcpy.AddMessage(f"{len(dsm_reg.putative_matches)} putative keypoint matches found.")
+        arcpy.AddMessage(f"{np.sum(dsm_reg.inliers)} keypoint matches found.")
+
+        feature_viz = os.path.join(dsm_reg.config["OUTPUT_DIR"], "dsm_feature_matches.png")
+        arcpy.AddMessage(f"Saving DSM feature match visualization to: {feature_viz}")
```

Comment:
Can you verify that the path is actually displayed correctly? The directory separators could act as escape sequences, but os.path _might_ handle that for us here... I'm not sure.

Reply:
### Verified! It saves as a valid path without backslash issues.

Reply:
yeah, the actual saving I knew would be fine, but the display I was a little concerned about.  If you see the path looking as it should that's good enough for me 👍🏻